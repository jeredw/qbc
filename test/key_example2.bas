CLS                       ' Clear screen
DIM KeyText$(3)
DATA Add, Delete, Quit
' Assign soft-key strings to F1 to F3.
FOR I = 1 TO 3
   READ KeyText$(I)
   KEY I, KeyText$(I) + CHR$(13)
NEXT I
' Print menu.
PRINT "                 Main Menu" : PRINT
PRINT "           Add to list (F1)"
PRINT "           Delete from list (F2)"
PRINT "           Quit (F3)" : PRINT
' Get input and respond.
DO
   LOCATE 7,1 : PRINT SPACE$(50);
   LOCATE 7,1 : INPUT "             Enter your choice:", R$
   SELECT CASE R$
      CASE "Add", "Delete"
         LOCATE 10,1 : PRINT SPACE$(15);
         LOCATE 10,1 : PRINT R$;
      CASE "Quit"
         EXIT DO
      CASE ELSE
         LOCATE 10,1 : PRINT "Enter first word or press key."
   END SELECT
LOOP
