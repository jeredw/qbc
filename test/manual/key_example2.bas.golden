(program 
  (statement 
    (call_statement 
      (untyped_id CLS))) \n 
  (statement 
    (dim_statement DIM 
      (dim_variable KeyText$ 
        (dim_array_bounds 
          ( 
            (dim_subscript 
              (expr 
                (literal 3))) ))))) \n 
  (statement 
    (data_statement DATA 
      (data_item Add) , 
      (data_item Delete) , 
      (data_item Quit))) \n \n 
  (statement 
    (for_next_statement FOR I = 
      (expr 
        (literal 1)) TO 
      (expr 
        (literal 3)) 
      (block \n 
        (statement 
          (read_statement READ 
            (variable_or_function_call KeyText$ 
              (args_or_indices 
                ( 
                  (expr 
                    (variable_or_function_call I)) ))))) \n 
        (statement 
          (key_statement KEY 
            (expr 
              (variable_or_function_call I)) , 
            (expr 
              (expr 
                (variable_or_function_call KeyText$ 
                  (args_or_indices 
                    ( 
                      (expr 
                        (variable_or_function_call I)) )))) + 
              (expr 
                (variable_or_function_call CHR$ 
                  (args_or_indices 
                    ( 
                      (expr 
                        (literal 13)) ))))))) \n) NEXT I)) \n \n 
  (statement 
    (print_statement PRINT 
      (expr 
        (literal "                 Main Menu")))) : 
  (statement 
    (print_statement PRINT)) \n 
  (statement 
    (print_statement PRINT 
      (expr 
        (literal "           Add to list 
          (F1)")))) \n 
  (statement 
    (print_statement PRINT 
      (expr 
        (literal "           Delete from list 
          (F2)")))) \n 
  (statement 
    (print_statement PRINT 
      (expr 
        (literal "           Quit 
          (F3)")))) : 
  (statement 
    (print_statement PRINT)) \n \n 
  (statement 
    (do_loop_statement DO 
      (block \n 
        (statement 
          (locate_statement LOCATE 
            (expr 
              (literal 7)) , 
            (expr 
              (literal 1)))) : 
        (statement 
          (print_statement PRINT 
            (expr 
              (variable_or_function_call SPACE$ 
                (args_or_indices 
                  ( 
                    (expr 
                      (literal 50)) )))) ;)) \n 
        (statement 
          (locate_statement LOCATE 
            (expr 
              (literal 7)) , 
            (expr 
              (literal 1)))) : 
        (statement 
          (input_statement INPUT "             Enter your choice:" , 
            (variable_or_function_call R$))) \n 
        (statement 
          (select_case_statement SELECT CASE 
            (expr 
              (variable_or_function_call R$)) 
            (before_first_case \n) 
            (case_block 
              (case_statement CASE 
                (case_expr 
                  (expr 
                    (literal "Add"))) , 
                (case_expr 
                  (expr 
                    (literal "Delete")))) 
              (block \n 
                (statement 
                  (locate_statement LOCATE 
                    (expr 
                      (literal 10)) , 
                    (expr 
                      (literal 1)))) : 
                (statement 
                  (print_statement PRINT 
                    (expr 
                      (variable_or_function_call SPACE$ 
                        (args_or_indices 
                          ( 
                            (expr 
                              (literal 15)) )))) ;)) \n 
                (statement 
                  (locate_statement LOCATE 
                    (expr 
                      (literal 10)) , 
                    (expr 
                      (literal 1)))) : 
                (statement 
                  (print_statement PRINT 
                    (expr 
                      (variable_or_function_call R$)) ;)) \n)) 
            (case_block 
              (case_statement CASE 
                (case_expr 
                  (expr 
                    (literal "Quit")))) 
              (block \n 
                (statement 
                  (exit_statement EXIT DO)) \n)) 
            (case_block 
              (case_statement CASE ELSE) 
              (block \n 
                (statement 
                  (locate_statement LOCATE 
                    (expr 
                      (literal 10)) , 
                    (expr 
                      (literal 1)))) : 
                (statement 
                  (print_statement PRINT 
                    (expr 
                      (literal "Enter first word or press key.")))) \n)) 
            (end_select_statement END SELECT))) \n) LOOP)) \n <EOF>)
